acceptance/tests/reports/corrective_change_new_resource.rb

```diff
@@ -0,0 +1,83 @@
+require 'yaml'
+
+test_name "C98092: corrective change new resource" do
+require 'puppet/acceptance/environment_utils'
+extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type = File.basename(__FILE__, '.*')
+  tmp_environment   = mk_tmp_environment_with_teardown(master, app_type)
+
+  tmp_file = {}
+  agents.each do |agent|
+    tmp_file[agent.hostname] = agent.tmpfile(tmp_environment)
+  end
+
+  teardown do
+    step 'clean out produced resources' do
+      agents.each do |agent|
+        on(agent, "rm #{tmp_file[agent.hostname]}", :accept_all_exit_codes => true)
+      end
+    end
+  end
+
+  file_contents     = 'this is a test'
+  step 'create site.pp with utf8 chars' do
+    manifest = <<MANIFEST
+file { '#{environmentpath}/#{tmp_environment}/manifests/site.pp':
+  ensure => file,
+  content => '
+\$test_path = \$::fqdn ? #{tmp_file}
+file { \$test_path:
+  content => @(UTF8)
+    #{file_contents}
+    | UTF8 
+}
+  ',
+}
+MANIFEST
+    apply_manifest_on(master, manifest, :catch_failures => true)
+  end
+
+  step 'run agent(s)' do
```

Comment:
Thanks for the valuable comments.Comments Incorporated

        #Run agent once to create new File resource
        on(agent, puppet("agent -t --environment #{tmp_environment} --server #{master.hostname}"),:acceptable_exit_codes => 2)

        #Verify the file resource is created
        on(agent, "cat #{tmp_file[fqdn]}").stdout do |file_result|
          assert_equal(file_contents, file_result, 'file contents did not match accepted')
        end

        #Delete the file
        on(agent, "rm #{tmp_file[fqdn]}", :accept_all_exit_codes => true)

        #Run agent to correct the file's absence
        on(agent, puppet("agent -t --environment #{tmp_environment} --server #{master.hostname}"),:acceptable_exit_codes => 2)

        #Verify the file resource is created
        on(agent, "cat #{tmp_file[fqdn]}").stdout do |file_result|
          assert_equal(file_contents, file_result, 'file contents did not match accepted')
        end